### ISSUE TITLE
Try a probit for a specific g and t.

### ISSUE THREAD

**Issue by zaveczgergo** [korenmiklos/expat-analysis#248]

**Comment by zaveczgergo** [comment]
`exporter` and `RperK` have all omitted.

**Comment by zaveczgergo** [comment]
I found this line in `attgt.ado`:
```
capture probit `TD' `ipw' if (`treated' | `control') & `touse' & (`time' == `g')
```
I am unsure how can I convert into a probit with a given `t` and `g` and a given `depvar` on a given `treatment`.

**Comment by korenmiklos** [comment]
Problem solved.

Attgt hardly works with `ipw` option, there are many omitted years. For many (g,t) pairs, there are only a few observations, and one cannot estimate an ipw probit at all.
```
note: exporter != 1 predicts failure perfectly
      exporter dropped and 13 obs not used

Iteration 0:   log likelihood = -3.8032073  
Iteration 1:   log likelihood =  -1.762594  
Iteration 2:   log likelihood =          0  
Iteration 3:   log likelihood =          0  

Probit regression                               Number of obs     =         17
                                                LR chi2(-1)       =       7.61
                                                Prob > chi2       =          .
Log likelihood =          0                     Pseudo R2         =     1.0000

------------------------------------------------------------------------------
    __00000J |      Coef.   Std. Err.      z    P>|z|     [95% Conf. Interval]
-------------+----------------------------------------------------------------
         lnQ |   1615.953          .        .       .            .           .
         lnK |  -463.8348          .        .       .            .           .
         lnL |  -491.8243          .        .       .            .           .
         lnM |  -668.3808          .        .       .            .           .
    exporter |          0  (omitted)
       _cons |  -8893.784          .        .       .            .           .
------------------------------------------------------------------------------
Note: 16 failures and 1 success completely determined.

```

See
```
. probit F3.expat_hire lnQ exporter if year==1995 & g==1997 

note: exporter != 1 predicts success perfectly
      exporter dropped and 8 obs not used

Iteration 0:   log likelihood = -3.0141613  
Iteration 1:   log likelihood =  -2.973875  
Iteration 2:   log likelihood = -2.9736642  
Iteration 3:   log likelihood = -2.9736642  

Probit regression                               Number of obs     =          8
                                                LR chi2(1)        =       0.08
                                                Prob > chi2       =     0.7760
Log likelihood = -2.9736642                     Pseudo R2         =     0.0134

-------------------------------------------------------------------------------
F3.expat_hire |      Coef.   Std. Err.      z    P>|z|     [95% Conf. Interval]
--------------+----------------------------------------------------------------
          lnQ |  -.1439804   .5071243    -0.28   0.776    -1.137926    .8499651
     exporter |          0  (omitted)
        _cons |   3.268794    7.52455     0.43   0.664    -11.47905    18.01664
-------------------------------------------------------------------------------
```

Only use the predicted propensity score if actually computed. Otherwise a weight of 1 will be used. (Why? Shouldn't it be +infty?)

```
. attgt lnQL, treatment(expat_hire) aggregate(e) pre(`pre') post(`post') notyet limitcontrol(foreign == 0) ipw(ln
> Q exporter)
       panel variable:  frame_id_numeric (unbalanced)
        time variable:  year, 1988 to 2018, but with gaps
                delta:  1 unit
Generating weights...

                          __000001
-------------------------------------------------------------
      Percentiles      Smallest
 1%     .0044525       6.65e-06
 5%      .009622       7.76e-06
10%     .0168481       8.39e-06       Obs              24,010
25%     .0374053       8.39e-06       Sum of Wgt.      24,010

50%            1                      Mean           .6189762
                        Largest       Std. Dev.      .8987531
75%            1       8.999998
90%            1       8.999998       Variance       .8077572
95%            1       8.999998       Skewness       6.569951
99%            1       8.999998       Kurtosis       62.14625
Estimating lnQL: event_m5
Estimating lnQL: event_m4
Estimating lnQL: event_m3
Estimating lnQL: event_m2
Estimating lnQL: event_m1
Estimating lnQL: event_1
Estimating lnQL: event_2
Estimating lnQL: event_3
Estimating lnQL: event_4
Estimating lnQL: event_5
Estimating lnQL: event_6
Estimating lnQL: event_7
Estimating lnQL: event_8
Estimating lnQL: event_9
Estimating lnQL: event_10
Callaway Sant'Anna (2021)

                                                Number of obs     =     15,953

------------------------------------------------------------------------------
             |      Coef.   Std. Err.      z    P>|z|     [95% Conf. Interval]
-------------+----------------------------------------------------------------
    event_m5 |    .030252   .0636403     0.48   0.635    -.0944807    .1549847
    event_m4 |   .0608625   .0478453     1.27   0.203    -.0329126    .1546376
    event_m3 |  -.0085453   .0430933    -0.20   0.843    -.0930066     .075916
    event_m2 |   .0142816   .0380744     0.38   0.708     -.060343    .0889061
    event_m1 |   .0130584   .0268679     0.49   0.627    -.0396017    .0657186
     event_1 |   .1210677   .0381834     3.17   0.002     .0462296    .1959059
     event_2 |   .2283719   .0488125     4.68   0.000     .1327012    .3240425
     event_3 |   .2734085   .0597507     4.58   0.000     .1562993    .3905177
     event_4 |   .2736761   .0685392     3.99   0.000     .1393418    .4080104
     event_5 |   .3128768   .0671692     4.66   0.000     .1812275    .4445261
     event_6 |   .3127892   .0691206     4.53   0.000     .1773154    .4482631
     event_7 |   .3179063   .0778686     4.08   0.000     .1652866    .4705261
     event_8 |   .3253324   .0881194     3.69   0.000     .1526216    .4980432
     event_9 |    .323338   .0937892     3.45   0.001     .1395145    .5071616
    event_10 |   .4113648   .1004354     4.10   0.000     .2145149    .6082146
------------------------------------------------------------------------------
```

With commit `a720a354a71b5fc14e3b66c05bae2f28bac60291` we have
```
. attgt lnQL, treatment(expat_hire) aggregate(e) pre(`pre') post(`post') notyet limitcontrol(foreign == 0) ipw(ln
> Q exporter)
       panel variable:  frame_id_numeric (unbalanced)
        time variable:  year, 1988 to 2018, but with gaps
                delta:  1 unit
Generating weights...

                          __000001
-------------------------------------------------------------
      Percentiles      Smallest
 1%     .0044525       6.65e-06
 5%      .009622       7.76e-06
10%     .0168481       8.39e-06       Obs              24,010
25%     .0371505       8.39e-06       Sum of Wgt.      24,010

50%            1                      Mean           .5456934
                        Largest       Std. Dev.      .4780567
75%            1              1
90%            1              1       Variance       .2285382
95%            1              1       Skewness      -.1065181
99%            1              1       Kurtosis       1.019798
Estimating lnQL: event_m5
Estimating lnQL: event_m4
Estimating lnQL: event_m3
Estimating lnQL: event_m2
Estimating lnQL: event_m1
Estimating lnQL: event_1
Estimating lnQL: event_2
Estimating lnQL: event_3
Estimating lnQL: event_4
Estimating lnQL: event_5
Estimating lnQL: event_6
Estimating lnQL: event_7
Estimating lnQL: event_8
Estimating lnQL: event_9
Estimating lnQL: event_10
Callaway Sant'Anna (2021)

                                                Number of obs     =     15,953

------------------------------------------------------------------------------
             |      Coef.   Std. Err.      z    P>|z|     [95% Conf. Interval]
-------------+----------------------------------------------------------------
    event_m5 |    .031589   .0633897     0.50   0.618    -.0926525    .1558304
    event_m4 |   .0634593   .0481819     1.32   0.188    -.0309756    .1578941
    event_m3 |  -.0087854   .0431164    -0.20   0.839    -.0932919    .0757211
    event_m2 |   .0138715   .0381447     0.36   0.716    -.0608907    .0886337
    event_m1 |   .0130248   .0268734     0.48   0.628    -.0396462    .0656958
     event_1 |   .1213298   .0382021     3.18   0.001     .0464551    .1962045
     event_2 |   .2285525   .0488551     4.68   0.000     .1327983    .3243067
     event_3 |     .27373   .0599242     4.57   0.000     .1562807    .3911792
     event_4 |   .2736571   .0687034     3.98   0.000      .139001    .4083133
     event_5 |   .3126736   .0672998     4.65   0.000     .1807684    .4445789
     event_6 |   .3134789   .0691299     4.53   0.000     .1779867    .4489711
     event_7 |   .3188368   .0779277     4.09   0.000     .1661013    .4715723
     event_8 |    .326462   .0880889     3.71   0.000      .153811     .499113
     event_9 |    .324693   .0936269     3.47   0.001     .1411878    .5081983
    event_10 |   .4132685   .1002576     4.12   0.000     .2167672    .6097698
------------------------------------------------------------------------------

```

Use patched version `v0.2.1`. There are indeed too few observations in many cases to estimate an ipw.
```
. attgt lnQL, treatment(expat_hire) aggregate(e) pre(`pre') post(`post') notyet limitcontrol(foreign == 0) ipw(ln
> Q exporter)
       panel variable:  frame_id_numeric (unbalanced)
        time variable:  year, 1988 to 2018, but with gaps
                delta:  1 unit
Generating weights...

                          __000001
-------------------------------------------------------------
      Percentiles      Smallest
 1%     .0042098              0
 5%     .0095481              0
10%      .016762              0       Obs              24,010
25%     .0370961              0       Sum of Wgt.      24,010

50%            1                      Mean           .5456385
                        Largest       Std. Dev.      .4781105
75%            1              1
90%            1              1       Variance       .2285896
95%            1              1       Skewness      -.1064753
99%            1              1       Kurtosis       1.019749
Estimating lnQL: event_m5
Estimating lnQL: event_m4
Estimating lnQL: event_m3
Estimating lnQL: event_m2
Estimating lnQL: event_m1
Estimating lnQL: event_1
Estimating lnQL: event_2
Estimating lnQL: event_3
Estimating lnQL: event_4
Estimating lnQL: event_5
Estimating lnQL: event_6
Estimating lnQL: event_7
Estimating lnQL: event_8
Estimating lnQL: event_9
Estimating lnQL: event_10
Callaway Sant'Anna (2021)

                                                Number of obs     =     15,596

------------------------------------------------------------------------------
             |      Coef.   Std. Err.      z    P>|z|     [95% Conf. Interval]
-------------+----------------------------------------------------------------
    event_m5 |    .031589   .0633897     0.50   0.618    -.0926525    .1558304
    event_m4 |          0  (omitted)
    event_m3 |          0  (omitted)
    event_m2 |   .0138715   .0381447     0.36   0.716    -.0608907    .0886337
    event_m1 |          0  (omitted)
     event_1 |   .1213298   .0382021     3.18   0.001     .0464551    .1962045
     event_2 |   .2285525   .0488551     4.68   0.000     .1327983    .3243067
     event_3 |     .27373   .0599242     4.57   0.000     .1562807    .3911792
     event_4 |   .2736571   .0687034     3.98   0.000      .139001    .4083133
     event_5 |   .3126736   .0672998     4.65   0.000     .1807684    .4445789
     event_6 |   .3134789   .0691299     4.53   0.000     .1779867    .4489711
     event_7 |   .3188368   .0779277     4.09   0.000     .1661013    .4715723
     event_8 |    .326462   .0880889     3.71   0.000      .153811     .499113
     event_9 |    .324693   .0936269     3.47   0.001     .1411878    .5081983
    event_10 |   .4132685   .1002576     4.12   0.000     .2167672    .6097698
------------------------------------------------------------------------------
```

This is especially bad for treatment3 (`expat_hire`). Maybe we could recycle the ipw from a joint treatment? If control looks like a good candidate for any of the three treatments, will be a good control.

Results are very similar with only lnQ used as predictor.

Version 0.2.2 looks great:
```
. attgt lnQL, treatment(expat_hire) aggregate(e) pre(5) post(10) notyet limitcontrol(foreign == 0) ipw(lnQ lnL ex
> porter)
       panel variable:  frame_id_numeric (unbalanced)
        time variable:  year, 1988 to 2018, but with gaps
                delta:  1 unit
Generating weights...

                          __000001
-------------------------------------------------------------
      Percentiles      Smallest
 1%     .0076865       2.42e-06
 5%     .0168684       .0000368
10%     .0211506       .0000379       Obs              24,010
25%     .0421806       .0000811       Sum of Wgt.      24,010

50%            1                      Mean           .5467481
                        Largest       Std. Dev.      .4771192
75%            1              1
90%            1              1       Variance       .2276428
95%            1              1       Skewness      -.1059831
99%            1       1.110199       Kurtosis       1.017615
Estimating lnQL: event_m5
Estimating lnQL: event_m4
Estimating lnQL: event_m3
Estimating lnQL: event_m2
Estimating lnQL: event_m1
Estimating lnQL: event_1
Estimating lnQL: event_2
Estimating lnQL: event_3
Estimating lnQL: event_4
Estimating lnQL: event_5
Estimating lnQL: event_6
Estimating lnQL: event_7
Estimating lnQL: event_8
Estimating lnQL: event_9
Estimating lnQL: event_10
Callaway Sant'Anna (2021)

                                                Number of obs     =     15,953

------------------------------------------------------------------------------
             |      Coef.   Std. Err.      z    P>|z|     [95% Conf. Interval]
-------------+----------------------------------------------------------------
    event_m5 |    .069418   .1242628     0.56   0.576    -.1741326    .3129686
    event_m4 |   .0789418   .1000424     0.79   0.430    -.1171377    .2750213
    event_m3 |   .0120859   .0892072     0.14   0.892    -.1627571    .1869289
    event_m2 |  -.0127604   .0696205    -0.18   0.855    -.1492142    .1236933
    event_m1 |  -.0230978   .0469768    -0.49   0.623    -.1151706     .068975
     event_1 |   .1240385   .0959583     1.29   0.196    -.0640364    .3121133
     event_2 |   .2317618   .1013556     2.29   0.022     .0331085    .4304151
     event_3 |   .2826139   .1136384     2.49   0.013     .0598867    .5053411
     event_4 |   .2708202   .1279055     2.12   0.034     .0201299    .5215104
     event_5 |   .3196657    .127459     2.51   0.012     .0698507    .5694807
     event_6 |   .3436098   .1401302     2.45   0.014     .0689597    .6182599
     event_7 |   .3440943   .1468191     2.34   0.019     .0563341    .6318546
     event_8 |   .3400767   .1634297     2.08   0.037     .0197605     .660393
     event_9 |   .3404361   .1797258     1.89   0.058      -.01182    .6926922
    event_10 |   .4131837   .1818285     2.27   0.023     .0568064    .7695609
------------------------------------------------------------------------------
```
